spec/rspec/support/differ_spec.rb

```diff
@@ -261,6 +261,35 @@ def inspect; "<BrokenObject>"; end
           expect(diff).to be_diffed_as(expected_diff)
         end
 
+        it "outputs unified diff message of two hashes with BigDecimal keys" do
+          in_sub_process_if_possible do
+            require 'bigdecimal'
+            expect {
+              diff = differ.diff({ BigDecimal('3') => 'b'}, { 3 => 'b' })
+              if expect(diff).to match(/3.0 \(#<BigDecimal/)
+                raise StandardError, 'Passed Test'
+              end
+            }.to raise_error(StandardError, 'Passed Test')
+          end
+        end
+
+        context 'with ActiveSupport' do
+          let(:date_time) { DateTime.new(2000, 1, 1, 1, 1, Rational(1, 10)) }
+
+          it "outputs unified diff message of two hashes with DateTime keys with the correct formatting" do
+            in_sub_process_if_possible do
+              require 'date'
+              stub_const("ActiveSupport", Module.new)
+              allow(date_time).to receive(:strftime).and_return("Timestamp")
+
+              expect {
+                diff = differ.diff({ date_time => 'b'}, { 3 => 'b' })
+                raise StandardError, 'Passed Test' if expect(diff).to match %r[Timestamp]
+              }.to raise_error(StandardError, 'Passed Test')
```

Comment:
Same here -- the "raise to signal a passing test" pattern is confusing with no benefit that I can see.